src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/WPFValueProvider.cs

```diff
@@ -26,35 +29,53 @@ public WPFValueProvider(IApplicationXamlFileAccessor applicationXamlFileAccessor
     }
 
     public override Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
     {
-        return GetPropertyValueAsync(propertyName);
+        return GetPropertyValueAsync(propertyName, defaultProperties);
     }
 
     public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
     {
-        await (propertyName switch
+        if (await IsWPFApplicationAsync(defaultProperties))
         {
-            StartupURIPropertyName => _applicationXamlFileAccessor.SetStartupUriAsync(unevaluatedPropertyValue),
-            ShutdownModePropertyName => _applicationXamlFileAccessor.SetShutdownModeAsync(unevaluatedPropertyValue),
+            await (propertyName switch
+            {
+                StartupURIPropertyName => _applicationXamlFileAccessor.SetStartupUriAsync(unevaluatedPropertyValue),
+                ShutdownModePropertyName => _applicationXamlFileAccessor.SetShutdownModeAsync(unevaluatedPropertyValue),
 
-            _ => throw new InvalidOperationException($"The {nameof(WPFValueProvider)} does not support the '{propertyName}' property.")
-        });
+                _ => throw new InvalidOperationException($"The {nameof(WPFValueProvider)} does not support the '{propertyName}' property.")
+            });
+        }
 
         return null;
     }
 
-    private async Task<string> GetPropertyValueAsync(string propertyName)
+    private async Task<string> GetPropertyValueAsync(string propertyName, IProjectProperties defaultProperties)
     {
-        return propertyName switch
+        if (await IsWPFApplicationAsync(defaultProperties))
         {
-            StartupURIPropertyName => await _applicationXamlFileAccessor.GetStartupUriAsync() ?? string.Empty,
-            ShutdownModePropertyName => await _applicationXamlFileAccessor.GetShutdownModeAsync() ?? "OnLastWindowClose",
+            return propertyName switch
+            {
+                StartupURIPropertyName => await _applicationXamlFileAccessor.GetStartupUriAsync() ?? string.Empty,
+                ShutdownModePropertyName => await _applicationXamlFileAccessor.GetShutdownModeAsync() ?? "OnLastWindowClose",
 
-            _ => throw new InvalidOperationException($"The {nameof(WPFValueProvider)} does not support the '{propertyName}' property.")
-        };
+                _ => throw new InvalidOperationException($"The {nameof(WPFValueProvider)} does not support the '{propertyName}' property.")
+            };
+        }
+
+        return string.Empty;
+    }
+
+    private static async Task<bool> IsWPFApplicationAsync(IProjectProperties defaultProperties)
+    {
+        string useWPFString = await defaultProperties.GetEvaluatedPropertyValueAsync(UseWPFPropertyName);
```

Comment:
This use be looking at the WPF capability to determine this - which can probably be provided to the class itself as an AppliesTo.